examples/guestbook/README.md

```diff
@@ -103,20 +128,96 @@ spec:
 [Download example](redis-master-controller.yaml?raw=true)
 <!-- END MUNGE: EXAMPLE redis-master-controller.yaml -->
 
-Change to the `<kubernetes>/examples/guestbook` directory if you're not already there. Create the redis master pod in your Kubernetes cluster by running:
+A Kubernetes [service](../../docs/user-guide/services.md) is a named load balancer that proxies traffic to one or more containers. This is done using the [labels](../../docs/user-guide/labels.md) metadata that we defined in the `redis-master` pod above.  As mentioned, we have only one redis master, but we nevertheless want to create a service for it. Why? Because it gives us a deterministic way to route to the single master using an elastic IP.
+
+Services find the pods to load balance based on the pods' labels.
+The selector field of the service description determines which pods will receive the traffic sent to the service, and the `port` and `targetPort` information defines what port the service proxy will run at.
+
+The file `examples/guestbook/redis-master-service.yaml` defines the redis master service:
+
+<!-- BEGIN MUNGE: EXAMPLE redis-master-service.yaml -->
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-master
+  labels:
+    app: redis
+    tier: backend
+    role: master
+spec:
+  ports:
+    # the port that this service should serve on
+  - port: 6379
+    targetPort: 6379
+  selector:
+    app: redis
+    tier: backend
+    role: master
+```
+
+[Download example](redis-master-service.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE redis-master-service.yaml -->
+
+First create the service by running:
+
+```console
+$ kubectl create -f examples/guestbook/redis-master-service.yaml
+service "redis-master" created
+```
+
+Then check the list of services, which should include the redis-master:
+
+```console
+$ kubectl get services
+NAME              CLUSTER_IP       EXTERNAL_IP       PORT(S)       SELECTOR                            AGE
+redis-master      10.0.136.3       <none>            6379/TCP      app=redis,role=master,tier=backend  1h
+```
+
+This will cause all pods to see the redis master apparently running on <ip>:6379.  A service can map an incoming port to any `targetPort` in the backend pod.  Once created, the service proxy on each node is configured to set up a proxy on the specified port (in this case port 6379).
+
+`targetPort` will default to `port` if it is omitted in the configuration. For simplicity's sake, we omit it in the following configurations.
+
+The traffic flow from slaves to masters can be described in two steps, like so:
+
+  - A *redis slave* will connect to "port" on the *redis master service*
+  - Traffic will be forwarded from the service "port" (on the service node) to the  *targetPort* on the pod that the service listens to.
+
+For more details, please see [Connecting applications](../../docs/user-guide/connecting-applications.md).
+
+#### Finding a service
```

Comment:
It's confusing to put this section title here. Please use something like this to separate each part:
`#### Define a service`
(The description of the service and its yaml file part)
`#### Create a service`
(The calling kubectl create part)
`#### Finding a service`
(The finding a service part)

This applies to the replication controller part too.